### PR TITLE
#6364 fixed sql, report template and removed warnings for null values

### PR DESCRIPTION
**Pull request by dragospodariu96**
#6364

### CODE REVIEW

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ReportStarter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metasfresh/metasfresh/pullRequest?prid=5242821&bid=17184836)

File: de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ReportStarter.java [hunk 1]
```diff
@@ -180,21 +180,8 @@ private void startProcessInvokeReportOnly(@NonNull final ReportPrintingInfo repo
 		//
 		// Set report data to process execution result
 		final ProcessExecutionResult processExecutionResult = processInfo.getResult();
-
+		final String reportFilename = result.getFilename() != null ? result.getFilename() : extractReportFilename(processInfo, outputType);
 		final String reportContentType = outputType.getContentType();
-
-		final String reportFilename;
```
Comment (metas-ts) [inline de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ReportStarter.java]: i kindof liked it this way (this version is also the fixed one). maybe it makes sense to extract it to a method like `extractResultFileName(result)`, but i particularly like the logging, for support purposes

File: de.metas.report/report-service/src/main/java/de/metas/report/rest/ReportRestController.java [hunk 2]
```diff
@@ -119,7 +97,7 @@ private static JsonReportError toJsonReportError(final Throwable throwable, fina
 	{
 		final Throwable cause = AdempiereException.extractCause(throwable);
 
-		final String adLanguageEffective = Check.isNotBlank(adLanguage)
+		final String adLanguageEffective = !Check.isBlank(adLanguage)
```
Comment (metas-ts) [inline de.metas.report/report-service/src/main/java/de/metas/report/rest/ReportRestController.java]: why?

File: de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ExecuteReportStrategy.java [hunk 3]
```diff
@@ -58,7 +58,7 @@ public static ExecuteReportResult of(@NonNull final OutputType outputType, @NonN
 
 		public static ExecuteReportResult of(@NonNull final String filename, @NonNull final OutputType outputType, @NonNull final byte[] reportData)
 		{
-			return new ExecuteReportResult(null, outputType, reportData);
+			return new ExecuteReportResult(filename, outputType, reportData);
```
Comment (metas-ts) [inline de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ExecuteReportStrategy.java]: yeah, that's the last fix that i made in x-mas and didn't yet put to master

Comment (dragospodariu96) [reply de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ExecuteReportStrategy.java]: yeah, thought it's good to put it with this ocassion!

File: de.metas.report/report-service/src/main/java/de/metas/report/rest/ReportRestController.java (hunk 2 above)
Comment (dragospodariu96) [reply de.metas.report/report-service/src/main/java/de/metas/report/rest/ReportRestController.java]: this somehow got here when merging the old branch into this one1

File: de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ReportStarter.java (hunk 1 above)
Comment (dragospodariu96) [reply de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ReportStarter.java]: makes sense to have logging, will revert to the old way! (Also this got changed when merging the old branch)

**Pull request merged**